lib/bindConnectors/bindMessage.js

```diff
@@ -76,11 +156,45 @@ module.exports = function (message, threadneedle, connectorConfig) {
 
 				threadneedle[methodName](body)
 
-				.done(resolve, reject);
+				.done(function (data) {
+					requestEnd(startTime, { connectorName, operationName: methodName, connectorVersion: connectorConfig.version, status: 'success' }, datadogInstance);
+					resolve(data);
+				}, function (data) {
+					if (process.env.ENABLE_LOGGING === 'true' && logger) {
+						logger.error(
+							data,
+							{
+								connectorName: connectorConfig.name,
+								connectorVersion: connectorConfig.version,
+								operation: event.header.message,
+								requestUrl: `${connectorConfig.globalModel.baseUrl(event.body)}${message.model.url}`,
+								workflowId: event.id,
+							}
+						);
+					}
+					requestEnd(startTime, { connectorName, operationName: methodName, connectorVersion: connectorConfig.version, status: 'failure' }, datadogInstance);
+					reject(data);
+				});
 
 			}
 
 		});
 	};
 
+	// Bind the method to the internal handler object - this is how the connector
+	// processes the inbound messages below.
+	return function (event) {
+		const startTime = performance.now();
+		const dataDogInitialisation = when(initDataDogApi());
+
+		return dataDogInitialisation.then(function ({ datadogApiInstance, datadogToken }) {
+			const logger = initialiseLogger(datadogToken);
+			return onFinishHandler(event, startTime, datadogApiInstance, logger);
+		}).catch(function (rejection) {
+			console.log(`something went wrong`, { rejection });
```

Comment:
The message here should be more verbose, e.g. `'Something went wrong with Datadog initilisation:'`

Reply:
we ended up removing this catch and handling failure to init the logger in a different way because it was causing operations to be executed twice